b2luigi/basf2_helper/utils.py

```diff
@@ -1,16 +1,18 @@
 import os
 
-import git
-
 
 def get_basf2_git_hash():
     basf2_release = os.getenv("BELLE2_RELEASE")
 
     if basf2_release == "head" or basf2_release is None:
-        basf2_release_location = os.getenv("BELLE2_LOCAL_DIR")
 
-        if basf2_release_location:
-            return git.Repo(basf2_release_location).head.object.hexsha
-        return "not_set"
+        try:
```

Comment:
Since I was here...

(In this way, `basf2_release` is equivalent to the `version` returned by `basf2 --info` or `basf2 --version`.)

Reply:
I the reasoning, but, in some sense, this is breaking backwards-compatibility from a user-perspective, as it changes the job output target path. That is, once updating, your jobs that were once _complete_ might suddenly not be _complete_ anymore. We're in beta anyway but I would probably change the second digit of the release for this and not just the thirds patch-digit.

I looked into the code and this seems to call the C++ function `getCommitID`:
https://github.com/belle2/basf2/blob/1c972b2c89ef11f38ee2f5ea8eb562dde0637155/framework/io/include/RootIOUtilities.h#L77-L84
> Return git SHA1 hashes taking into account local & central release.

This could have been a separate PR. Fine with me having it here, but I would make this a prominent line in the `Changed` section of the Changelog

Reply:
I was about to merge but found that this change is breaking unittests due to failed import of `basf2`, see https://github.com/nils-braun/b2luigi/actions/runs/4595212940/jobs/8115187454?pr=193.

I think this is because previously, if no basf2 release is set, `get_basf2_git_hash` just returned the string `"not set"` if no release is set. You can argue in favor of raising an error then, but then we would need to fix the unit tests. This might be done by using `unittest.Mock` to mock a fake `basf2.version` package with a `get_version` function or mock the entire `get_basf2_git_hash` function.

- [ ] Fix unittests for `get_basf2_git_hash`, either return `"not set"` in case of failure or fix unittests by using mocking.

Reply:
I now reverted the behavior to returning `"not set"` when basf2 can't be imported in 0180a725ed8d2ee6ae7f8b2f5a5f13dbfbe39419, but now additionally print a warning because that's really undesired behavior usually. I also set up unit tests which document the functionality of `get_basf2_git_hash` in 374944c69bf3d5acce702bfba89d107835aee679 andbeaa520f688708e74cf5f0a24726c4804af2807e